### ISSUE TITLE
Karma running VERY slowly and running twice per build

### ISSUE THREAD

**Issue by dillonforrest** [ngbp/ngbp#228]
For some unknown reason, my karma runner started getting extremely slow.  I hit 120 seconds to run just 5 assertions, and 15 seconds to run just 2.  Here's the output:

```
>> File "src/app/app.js" changed.

Running "jshint:src" (jshint) task
>> 7 files lint free.

Running "karma:unit:run" (karma) task
[2014-02-25 19:12:49.927] [DEBUG] config - Loading config /Users/dillonforrest/campfire/build/karma-unit.js
....
Chrome 33.0.1750 (Mac OS X 10.8.5): Executed 2 of 2 SUCCESS (14.709 secs / 7.516 secs)
Chrome 33.0.1750 (Mac OS X 10.8.5): Executed 2 of 2 SUCCESS (0.153 secs / 0.05 secs)
TOTAL: 4 SUCCESS

Running "copy:build_appjs" (copy) task
Copied 7 files

Done, without errors.
Completed in 15.799s at Tue Feb 25 2014 19:13:04 GMT-0500 (EST)
```

15.799s was NOT typical before today, with only those two assertions.  I was getting easily less than 1s as recent as this morning.  I have NO idea why it suddenly slowed up so much, or why it hasn't gone back to its speed even at previous commits.

My `karma/karma-unit.tpl.js` looks like this:

``` javascript
module.exports = function ( karma ) {
  karma.set({
    /**
     * From where to look for files, starting with the location of this file.
     */
    basePath: '../',

    /**
     * This is the list of file patterns to load into the browser during testing.
     */
    files: [
      <% scripts.forEach( function ( file ) { %>'<%= file %>',
      <% }); %>
      'src/**/*.js',
    ],
    exclude: [
      'src/assets/**/*.js'
    ],
    frameworks: [ 'jasmine' ],
    plugins: [ 'karma-jasmine', 'karma-chrome-launcher' ],

    /**
     * How to report, by default.
     */
    reporters: 'dots',

    /**
     * On which port should the browser connect, on which port is the test runner
     * operating, and what is the URL path for the browser to use.
     */
    port: 9018,
    runnerPort: 9100,
    urlRoot: '/',

    /**
     * Disable file watching by default.
     */
    autoWatch: false,

    /**
     * The list of browsers to launch to test on. This includes only "Firefox" by
     * default, but other browser names include:
     * Chrome, ChromeCanary, Firefox, Opera, Safari, PhantomJS
     *
     * Note that you can also use the executable name of the browser, like "chromium"
     * or "firefox", but that these vary based on your operating system.
     *
     * You may also leave this blank and manually navigate your browser to
     * http://localhost:9018/ when you're running tests. The window/tab can be left
     * open and the tests will automatically occur there during the build. This has
     * the aesthetic advantage of not launching a browser every time you save.
     */
    browsers: [
      'Chrome'
    ]
  });
};
```

The first time `grunt watch` runs, the tests run perfectly fine.  But after that, the tests seem to run twice, and the first time is the time which takes up almost all the time.  See the above karma output for example.

Does anybody have any thoughts about this?

**Comment by PaulL1** [comment]
I use Chrome, and I find that if the tab it's running in is not in focus then it runs very very slowly. It's ok if it's the front tab in another window, but not if it's the second or third tab in a given window.

**Comment by joshdmiller** [comment]
@PaulL1 I have the exact same problem in Chrome on Linux across all projects, ngbp and not.

**Comment by PaulL1** [comment]
Agree.  I suspect it's a deliberate optimisation within Chrome to prevent background tabs consuming all the resource in running some crazy javascript process.  I usually open an additional Chrome window with only the single Karma tab within it, avoiding having to remember to always bring it back to the front.

**Comment by dillonforrest** [comment]
Hey guys!  I think I found the problem.  Like @PaulL1 said, Chrome ran the tests very slowly when the test tab wasn't in focus, but it worked fine in a new window.  My problem was that I had the original test tab in a separate window, but then I opened a second tab in my main Chrome window so I could use the jasmine debugger.  That second page was slowing me down!!  So that explains both the slowness and the double testing.  Thanks guys!!  :)
